hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/storefiletracker/StoreFileTrackerBase.java

```diff
@@ -98,7 +101,10 @@ public final void add(Collection<StoreFileInfo> newFiles) throws IOException {
   @Override
   public final void replace(Collection<StoreFileInfo> compactedFiles,
     Collection<StoreFileInfo> newFiles) throws IOException {
-    if (isPrimaryReplica) {
+    if (
+      isPrimaryReplica && !conf.getBoolean(HConstants.HBASE_GLOBAL_READONLY_ENABLED_KEY,
+        HConstants.HBASE_GLOBAL_READONLY_ENABLED_DEFAULT)
+    ) {
```

Comment:
Can we add same check to the `add` and `set` method? It would give an extra safeguard just in case these methods are inadvertently called on read-replica.